### PR TITLE
Adapting IoT docs to the new registry API and doing some small imrpov…

### PR DESCRIPTION
**Pull request by dejanb**
…ements

### CODE REVIEW

File: documentation/modules/proc-iot-api.adoc [hunk 1]
```diff
@@ -27,23 +27,12 @@ endif::[]
 
 .Procedure
 
-. In the Hono project, run:
-+
-[options="nowrap",subs="attributes"]
-----
-cd cli
-----
-. Run the following command at least once:
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-api.adoc]: What does "at least once" mean? Assuming this step is to install the command-line client, suggest changing to: "Run the following command to install the command-line client:"

File: documentation/modules/proc-iot-creating-device.adoc [hunk 2]
```diff
@@ -21,17 +21,31 @@ endif::[]
...
-. Register a device:
+. Export the device registry host:
 +
 [options="nowrap",subs="attributes"]
 ----
-curl --insecure -X POST -i -H 'Content-Type: application/json' --data-binary '{"device-id": "4711"}' https://{registry}/registration/myapp.iot
+export REGISTRY_HOST={registry}
+----
+
+. Register a device with a defined id (in this example we're using `4711`):
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-creating-device.adoc]:
```suggestion
. Register a device with a defined ID (in this example we're using `4711`):
```

File: documentation/modules/proc-iot-creating-device.adoc [hunk 3]
```diff
@@ -21,17 +21,31 @@ endif::[]
...
 . Add credentials for a device:
 +
 [options="nowrap",subs="attributes"]
 ----
-curl --insecure -X POST -i -H 'Content-Type: application/json' --data-binary '{"device-id": "4711","type": "hashed-password","auth-id": "sensor1","secrets": [{"hash-function" : "sha-512","pwd-plain":"'hono-secret'"}]}' https://{registry}/credentials/myapp.iot
+curl --insecure -X PUT -i -H 'Content-Type: application/json' --data-binary '[{
+	"type": "hashed-password",
+	"auth-id": "sensor1",
+	"secrets": [{
+		"hash-function" : "sha-512",
```
Comment (ctron) [inline documentation/modules/proc-iot-creating-device.adoc]: IIRC the hash-function should not be present when putting in a plain password, as the server side decides the hash function.

Comment (ctron) [reply documentation/modules/proc-iot-creating-device.adoc]: Maybe also add an explanation about the concept of handling/storing hashed passwords.

File: documentation/modules/proc-iot-api.adoc (hunk 1 above)
Comment (dejanb) [reply documentation/modules/proc-iot-api.adoc]: This is now removed from the doc. It meant that you need to compile Hono command line source code at least once. With this PR we're downloading binary client, so no need for this anymore.

File: documentation/modules/proc-iot-creating-device.adoc (hunk 3 above)
Comment (dejanb) [reply documentation/modules/proc-iot-creating-device.adoc]: Yeah, I'll add another PR that have more info around all topics. So it'll be covered there.

**Pull request merged**